client/src/cmdlfem4x50.c

```diff
@@ -779,6 +781,12 @@ static int CmdEM4x50Reader(const char *Cmd) {
         // iceman,  misuse of return status code.
         int now = resp.status;
 
+        // prevent massive stack corruption if unexpected results from device.
+        if (now > EM4X50_NO_WORDS) {
+            PrintAndLogEx(WARNING, "word count was: %d, limiting to %d", now, EM4X50_NO_WORDS);
+            now = EM4X50_NO_WORDS;
+        }
+
```

Comment:
the proper fix would be not misusing the return status code and check it accordingly

Reply:
Yes, that would be the proper fix.
This PR is focused on CodeQL fixes.  It is not intended to fix other problems.  Thus, this is outside the scope of this PR.